src/Authentication.Abstractions/AzureSession.cs

```diff
@@ -229,7 +231,12 @@ public static void Modify(Action<IAzureSession> modifier)
                     var key = new ComponentKey(componentName, typeof(T));
                     if (!_componentRegistry.ContainsKey(key) || overwrite)
                     {
-                        _componentRegistry[key] = componentInitializer();
+                        var component = componentInitializer();
```

Comment:
We should remove listener in the case _componentRegistry.ContainsKey(key) && overwrite

Reply:
this logic was covered by || overwrite isn't it?

Reply:
> We should remove listener in the case _componentRegistry.ContainsKey(key) && overwrite

this logic was covered by || overwrite isn't it?